cpp/tests/community/louvain_test.cu

```diff
@@ -62,20 +62,32 @@ TEST(louvain, success)
 
   raft::handle_t handle;
 
-  std::tie(num_level, modularity) = cugraph::louvain(handle, G, result_v.data().get());
-
-  cudaMemcpy((void*)&(cluster_id[0]),
-             result_v.data().get(),
-             sizeof(int) * num_verts,
-             cudaMemcpyDeviceToHost);
-
-  int min = *min_element(cluster_id.begin(), cluster_id.end());
-
-  std::cout << "modularity = " << modularity << std::endl;
-
-  ASSERT_GE(min, 0);
-  ASSERT_GE(modularity, 0.402777 * 0.95);
-  ASSERT_EQ(result_v, result_h);
+  // "FIXME": remove this check once we drop support for Pascal
+  //
+  // Calling louvain on Pascal will throw an exception, we'll check that
+  // this is the behavior while we still support Pascal (device_prop.major < 7)
+  //
+  cudaDeviceProp device_prop;
+  CUDA_CHECK(cudaGetDeviceProperties(&device_prop, 0));
+
+  if (device_prop.major < 7) {
```

Comment:
Should this test file be .cu?

Our users are C++ and I think we should allow C++ users using g++ to use libcugraph, and to test that, I think our C++ files better be .cpp than .cu.

Reply:
And better use rmm::device_uvector instead of rmm::device_vector. rmm::device_uvector compiles with g++. rmm:device_vector is a thrust::device_vector and requires nvcc.

Reply:
Renamed all of the Louvain derived tests to `.cpp`.  Modified tests to use `rmm::device_uvector`.